src/sass/components/_grid.scss

```diff
@@ -1,170 +1,174 @@
-  'junior': 1140px,
-  'senior': 1380px
+// These are the max-widths of the row. Feel free to add more
+// sizes/key: value pairs that can then be used on the .row container
+// like, .row .row--yournewsize.
+
+$row-widths: (
+  "sm": 640px,
+  "md": 1024px,
+  "lg": 1140px,
+  "xl": 1380px
 );
 
-// Global grid gutter. Set this to half of what you want the final gutter to be.
-$gutter: $md / 2;
+// Global row gutter. Set this to half of what you want the final gutter to be.
+$gutter: $md/2;
 
-// Starting on some mixins to build grid classes and to make
+// Starting on some mixins to build row classes and to make
 // available for those who want to use the Sass source
 // source files in their app.
 
 @mixin container($width: 100%) {
+  margin-left: auto !important;
+  margin-right: auto !important;
   max-width: $width;
-  padding-right: $gutter * 2;
   padding-left: $gutter * 2;
+  padding-right: $gutter * 2;
 }
 
-@mixin grid {
+@mixin row {
   display: flex;
   flex-wrap: wrap;
-  margin-right: ($gutter / -1);
-  margin-left: ($gutter / -1);
-}
-
-.#{$prefix}-container {
-  @include container();
+  margin: 0 ($gutter/-1);
 }
 
 /**
- * Contianer variants
+ * Container variants
  */
 
-.#{$prefix}-container--center {
-  margin-right: auto !important;
-  margin-left: auto !important;
-}
-
-@each $key, $value in $grid-widths {
-  .#{$prefix}-container--#{$key} {
-    max-width: $value;
+@each $key, $value in $row-widths {
+  .#{$prefix}-container-#{$key} {
+    @include container($value);
   }
 }
 
-.#{$prefix}-container--collapse {
-  padding: 0;
-}
-
 /**
- * The main grid container. It's width is fluid by default.
+ * The main row container. It's width is fluid by default.
  */
 
-// TODO: refactor the other grid__item-* classes into mixins for those
-// that want ot use the Sass source in their app.
+// TODO: refactor the other row__cols-* classes into mixins for those
+// that want to use the Sass source in their app.
 
-@mixin grid-item {
-  max-width: 100%;
+@mixin cols {
   flex-basis: 0;
   flex-grow: 1;
+  max-width: 100%;
   padding: 0 $gutter;
   position: relative;
 }
 
-.#{$prefix}-grid {
-  @include grid;
+.#{$prefix}-row {
+  @include row;
+
+  &--loose {
+    margin: 0 ($gutter * -1.66);
+  }
+
+  &--tight {
+    margin: 0 ($gutter * -.75);
+  }
 
-  &__item {
-    @include grid-item;
+  .#{$prefix}-cols {
+    @include cols;
   }
 
-  &__item--last {
+  .#{$prefix}-cols--last {
     margin-left: auto;
   }
 
-  [class^='#{$prefix}-grid__item'] {
+  [class^="#{$prefix}-cols"] {
     /**
-     * This keeps the grid from collapsing when items inside are too
-     * big for their parent .#{$prefix}-grid__item.
+     * This keeps the row from collapsing when cols inside are too
+     * big for their parent .#{$prefix}-row__cols.
      */
     min-width: 0;
   }
+
+  &--loose > [class^="#{$prefix}-cols"] {
+    padding: 0 ($gutter * 1.66);
+  }
+
+  &--tight > [class^="#{$prefix}-cols"] {
+    padding: 0 ($gutter * .75);
+  }
```

Comment:
This is kind of nit-picky, but can you move these two modifier rules up so they come right after the original `--tight` and `--loose` declarations? It's a little easier to understand what the modifiers are doing if you can see the styles they're applying in close proximity to each other.